torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -584,7 +584,8 @@ ProcessGroupNCCL::ProcessGroupNCCL(
       "ProcessGroupNCCL is only supported with GPUs, no GPUs found!");
   blockingWait_ = parseEnvVarFlag(NCCL_BLOCKING_WAIT);
   asyncErrorHandling_ = parseEnvVarFlag(NCCL_ASYNC_ERROR_HANDLING);
-  desyncDebug_ = parseEnvVarFlag(NCCL_DESYNC_DEBUG);
+  desyncDebug_ = parseEnvVarFlag(NCCL_DESYNC_DEBUG) ||
+      dist_debug_level_ >= DebugLevel::Detail;
```

Comment:
Nit: I think PyTorch allows up to 120 character long strings. 
Also, use brackets, as operator precedence can be tricky in those cases
```suggestion
  desyncDebug_ = parseEnvVarFlag(NCCL_DESYNC_DEBUG) || (dist_debug_level_ >= DebugLevel::Detail);
```